### PR TITLE
Allow passthrough redirects

### LINKED ISSUE #124: fastify-http-proxy does not respect 302 redirects
## 🐛 Bug Report
Hey, thanks for this amazing project. I have some trouble when the upstream servers are triggering some redirects. It seems that the proxy is prefixing the 

## To Reproduce

```js
const Fastify = require('fastify')
const proxy = require('fastify-http-proxy')

const server = Fastify()
const upstream = Fastify()


// Declare a route
const Fastify = require('fastify');
const proxy = require('fastify-http-proxy');

const server = Fastify();
const upstream = Fastify();

// Declare a route
upstream.get('/', async (request, reply) => {
  return reply.redirect(302, 'https://google.com');
});

upstream.get('/foo', async (request, reply) => {
  return reply.send({hello: 'world'});
});

upstream.listen(3001);

server.register(proxy, {
  upstream: 'http://localhost:3001',
  prefix: '/bar'
});

server.listen(3000);

```

## Expected behavior

Visiting `http://localhost:3000/bar/foo` the proxy works as expected, visiting `http://localhost:3000/bar` a new redirect was triggered to `/barhttps://google.com`

## Your Environment

- *node version*: 14
- *fastify version*: 3.9.2
- *os*: Mac

### PR DESCRIPTION
**Pull request by vorillaz**
#### Description
Allow absolute URL redirects when applicable. Closes #124 .

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -125,7 +135,8 @@ async function httpProxy (fastify, opts) {
 
   function rewriteHeaders (headers) {
     const location = headers.location
-    if (location) {
+    console.log(location)
```
Comment (mcollina) [inline index.js:138]: please remove the console.log.

File: index.js [hunk 2]
```diff
@@ -31,6 +32,15 @@ function waitConnection (socket, write) {
   }
 }
 
+function isExternalUrl (url) {
+  try {
+    const u = new URL(url)
```
Comment (mcollina) [inline index.js:37]: unfortunately this is going to be very costly perf-wise. new URL is slow. I would recommend to use a simple regexp matching http and https prefixes.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (vorillaz) [reply index.js]: Mea culpa, fixed.

File: index.js (hunk 2 above)
Comment (vorillaz) [reply index.js]: Got it, fixed as well.

File: index.js [hunk 3]
```diff
@@ -31,6 +31,10 @@ function waitConnection (socket, write) {
   }
 }
 
+function isExternalUrl (url = '') {
+  return urlPattern.test(url)
```
Comment (Eomm) [inline index.js:35]: Would you mind to check if `url.startsWith('http://') || url.startsWith('https://')` is faster than regex?

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by vorillaz** [comment]
Awesome, happy new year @mcollina. Waiting for this one getting merged :D

File: index.js (hunk 3 above)
Comment (jsumners) [inline index.js:35]: How is this guaranteed to be "an external url"? Shouldn't the headers be inspected to determine if there is a `3xx` code that should be honored?

Comment (vorillaz) [reply index.js:35]: That was my initial thought but `status code` is not getting passed through the `rewriteHeaders` method

Comment (jsumners) [reply index.js:35]: Then I would think the solution starts with resolving that issue. I do not see how this simple test is a guarantee of a redirect.

**Comment by mcollina** [comment]
any updates on this?

**Comment by vorillaz** [comment]
I am not sure I can merge this one as there are no action items, I resolved all the issues but merging is blocked and I don't know why.

**Comment by jsumners** [comment]
> I am not sure I can merge this one as there are no action items, I resolved all the issues but merging is blocked and I don't know why.

Merging is not blocked, per se. I would certainly like my question addressed, but I'm not blocking this.

File: index.js (hunk 3 above)
Comment (climba03003) [reply index.js:35]: If `status code` is necessary for the checking, I think you can start a PR in [`fastify-reply-form`](https://github.com/fastify/fastify-reply-from) to achieve it.

**Comment by csvan** [comment]
@vorillaz could you try merging this again if nothing is blocking?

**Pull request merged**